src/field/pokemon.ts

```diff
@@ -1306,6 +1283,38 @@ export default abstract class Pokemon extends Phaser.GameObjects.Container {
         if (types.length === 1 && speciesForm.type2 !== null) {
           types.push(speciesForm.type2);
         }
+
+        // "Permanent" override for a Pokemon's normal types, currently only used by Mystery Encounters
+        if (this.customPokemonData.types && this.customPokemonData.types.length > 0) {
```

Comment:
```suggestion
        if (this.customPokemonData.types?.length > 0) {
```
Pretty sure you can simplify it to this.